irods/test/resource_test.py

```diff
@@ -0,0 +1,85 @@
+#! /usr/bin/env python
+from __future__ import absolute_import
+import os
+import six
+import sys
+import unittest
+
+from irods.test import helpers
+
+class TestResource(unittest.TestCase):
+
+    from helpers import create_simple_resc_hierarchy
+
+    def setUp(self):
+        self.sess = helpers.make_session()
+
+    # this will probably replace the older test (which follows below it, and tested only at 2 levels)
+
+    def test_resource_properties_for_parent_and_hierarchy_at_3_levels__392(self):
+        ses = self.sess
+        root = 'deferred_392'
+        pt = 'pt_392'
+        leaf = 'leaf_392'
+        root_resc = ses.resources.create(root,'deferred')
+        try:
+            # Create two (passthru + storage) hierarchies below the root: ie. pt0;leaf0 and pt1;leaf1
+            with self.create_simple_resc_hierarchy(pt + "_0",leaf + '_0') as resc_hier_0, \
+                 self.create_simple_resc_hierarchy(pt + "_1",leaf + '_1') as resc_hier_1:
+                try:
+                    # Adopt both passthru's as children under the main root (deferred) node.
+                    ses.resources.add_child( root, pt + "_0")
+                    ses.resources.add_child( root, pt + "_1")
+                    # Now we have two different 3-deep hierarchies (root;pt0;leaf0 and root;pt1;leaf) sharing the same root node.
+                    # Descend each and make sure the relationships hold
+                    for mid in root_resc.children:
+                        hierarchy = [root_resc, mid, mid.children[0]]
+                        parent_resc = None
+                        hier_str = root
+                        # Assert that the hierarchy and parent properties hold at each level, in both tree branches.
```

Comment:
Is the idea here that we are testing `.hierarchy` from the root resource, middle resource(s), and leaf resource(s)?

Reply:
yes, exactly.